### PR TITLE
Add pagination for GET /api/v1/assets

### PR DESCRIPTION
**Pull request by kraih**
Progress: https://progress.opensuse.org/issues/120841

### CODE REVIEW

**Comment by kraih** [comment]
Tests will pass once Mojolicious 9.30 is available here.

File: lib/OpenQA/WebAPI/Controller/API/V1/Asset.pm [hunk 1]
```diff
@@ -59,15 +59,39 @@ as its id, name, timestamp of creation and type is included.
...
+
+    # Pagination
+    pop @all if my $has_more = @all > $limit;
+    $self->_pagination_links($limit, $offset, $has_more);
+
+    $self->render(json => {assets => \@all});
+}
+
+# This could be turned into a helper if other API endpoints want to use it
+sub _pagination_links ($self, $limit, $offset, $has_more) {
```
Comment (Martchus) [inline lib/OpenQA/WebAPI/Controller/API/V1/Asset.pm:87]:
```suggestion
sub _set_pagination_links_header ($self, $limit, $offset, $has_more) {
```

**Review by Martchus — COMMENTED** [review]
Looks generally very good.

**Comment by kraih** [comment]
Test passed, ready for review.

**Review by okurz — APPROVED** [review]
Very nice. I suggest to await some days to give more distributions to provide the updated Mojolicious framework

**Review by Martchus — APPROVED** [review]

**Comment by kraih** [comment]
It has been some days. 😉

**Pull request merged**